### ISSUE TITLE
Security key authentication flow issue [iOS 14]

### ISSUE THREAD

**Issue by Uni-verse** [brave/brave-ios#5100]
### Description: 
When going through the 2FA Security key flow, there is a modal prompting to launch external link. After proceeding, the flow stalls with infinite loader (see video)

- **Does not impact iOS 15.X** 
- **User is able to successfully authenticate using the same repro steps in Safari Browser**

### Steps to Reproduce 
  1. Navigate to Google.com
  2. Click to Sign in
  3. Enter credentials, tap next
  4. On the 2 step verification screen, tap next 
  5. Tap 'Allow' at the modal prompt
  6. Insert security key and contact to authenticate

**Actual result:** 
Authentication flow stuck on 2 Step Verification screen after dismissing modal

https://user-images.githubusercontent.com/17885425/158260656-6270c7a3-f7dc-47cb-9cd5-0b9c2326f8c3.MP4

**Expected result:**
Modal should not be appear in flow, user should successfully be able to authenticate with yubikey

**Reproduces how often:** [Easily reproduced, Intermittent Issue]
Easily, 100% repro rate

**Brave Version:** 
Version 1.36 (22.3.11.16)

**Device details:** 
iPhone X | iOS 14.4.2

**Website problems only:**
- did you check with Brave Shields down?
- did you check in Safari/Firefox (WkWebView-based browsers)?

**Comment by kjozwiak** [comment]
hmm seems to be working on my device. Maybe it has something to do with a specific Yubikey version? Either way, here's all the device information and STR/Cases I used:

* `iPhone 11` running `iOS 14.7.1` using `1.36 (22.3.11.16)`
* `Yubikey 4 FW: 4.3.7`

STR/Cases:

* added the yubikey into my Google account via `1.38.39 Chromium: 99.0.4844.51`
* once added, installed `1.36 (22.3.11.16)` and loaded `gmail.com`
* used `1Password` to autofill in the password and was presented with the 2FA screen
* plugged in the yubikey that was added previously and clicked on `Next` via Gmail
* pressed the two metal connectors on the key was logged into Gmail without any issues

**`Example`** | **`Example`** | **`Example`**
---------------|----------------|-----------------
![image](https://user-images.githubusercontent.com/2602313/158264236-5d6cc40f-1493-460f-a530-196902850b73.png) | ![image](https://user-images.githubusercontent.com/2602313/158264315-4673c93a-281e-4eda-a021-ab02e22c137d.png) | ![image](https://user-images.githubusercontent.com/2602313/158264457-7a82b67a-3175-4f54-b34a-a8a182690650.png)

**Comment by kjozwiak** [comment]
@Brandon-T just curious, looks like @Uni-verse is using an older `iOS 14`. Does `iOS 14.4.2` support https://github.com/brave/brave-ios/pull/4758?

**Comment by Uni-verse** [comment]
Could the key itself be the reason? I think i have the non-NFC yubikey.

**Comment by Brandon-T** [comment]
This seems to be a pop up we have for external URLs that is causing it. It's almost as if Google is attempting to open the Authenticator application which has that URL scheme. So we probably have to allow that specific scheme through. Not sure if we should through.

@iccub @anthonypkeane

**Comment by kjozwiak** [comment]
> This seems to be a pop up we have for external URLs that is causing it. It's almost as if Google is attempting to open the Authenticator application which has that URL scheme. So we probably have to allow that specific scheme through. Not sure if we should through.
> 
> @iccub @anthonypkeane

Makes sense as I don't have the authenticator app enabled on my account hence not being able to reproduce. @Uni-verse assuming it's enabled on your account?

**Comment by Uni-verse** [comment]
@kjozwiak Actually its not enabled. I didn't even have the app installed before I ran into this issue.

**Comment by Brandon-T** [comment]
Hi, this issue cannot be fixed on our end.
I was able to reproduce it only on devices that do `not` support `WebAuthN` such as `simulator` for example.

On an actual device like the iPhone 12 Pro, it attempts to do WebAuthN, and if it is supported, the user gets the prompt normally. If it isn't supported, `Google` is actually making an attempt to open their own app to do it. This attempt fails if the app is not installed, and so the user can never log in anyway. 

Unfortunately, there isn't any way for us to tell websites that the attempt to open some arbitrary URL won't work.
Below are what I see: 
<img width="342" alt="image" src="https://user-images.githubusercontent.com/1530031/160669527-d58edbe1-2ac8-495d-a74f-3bfc4802b00e.png"> <img width="376" alt="image" src="https://user-images.githubusercontent.com/1530031/160669603-8a26e9a8-3925-49f2-a03c-2754c7705a00.png">

If we attempt to help gmail by allowing the URL to go through automatically, we will get `not supported URL` page anyway.
I tested on Safari, and we have the same behaviour (doing nothing and letting the page load infinitely, as it isn't possible to check if the user has the app installed first).
